components/movementsensor/movementsensor.go

```diff
@@ -185,8 +186,9 @@ func Readings(ctx context.Context, g MovementSensor, extra map[string]interface{
 	if err != nil && !errors.Is(err, ErrMethodUnimplementedOrientation) {
 		return nil, err
 	}
-	readings["orientation"] = ori
-
+	if ori != nil {
```

Comment:
curious: why do we need this check for orientation and not for other types?
I wonder if:
1. we need a similar check for all others
2. if our drivers are inconsistent wrt what they return if there's no data to return...e.g. nil vs. some default value.

Reply:
1. No because the other readings call the constructor and default to values like `0` for floats or `lat = 0, lng = 0` for points
2. It looks like only the `orientation` function in components/movementsensor/gpsnmea/serial.go  returns `nil` https://github.com/viamrobotics/rdk/blob/6f6e6bbb17970bf005cb4b5e4b13f4db2d59aa35/components/movementsensor/gpsnmea/serial.go#L191 would you prefer I return a default `spatialmath.Orientation` to be consistent with the other readings or would a default orientation be misleading?

Reply:
Updated for all movement sensors that previously returned nil and verified it does not break sensor card on app with gpsnmea